lib/vdc/BOVCollection.cpp

```diff
@@ -181,6 +197,20 @@ int BOVCollection::_parseHeader(std::ifstream &header)
 
 int BOVCollection::_validateParsedValues()
 {
+    // Validate that the data file exists.
+    // If not given an absolute path, construct one
+    // from the header file's containing directory.
+    char  actualPath[PATH_MAX + 1];
+    char *success = realpath(_dataFile.c_str(), actualPath);
+    if (success == nullptr) {
+        // At this point we can't find the absolute path, so it might be a relative path.
+        // Try prepending the directory of the .bov file to the data file
+        _dataFile = _currentFilePath + "//" + _dataFile;
```

Comment:
@sgpearse, we have a collection of filesystem path manipulation routines in FileUtils.{h,cpp} that provide OS agnostic support for path concatenation, etc. We should use these - and possibly extend as necessary - rather than reinventing the wheel here (and above).

That said, c++17 provides (finally!!) STL utilities for path manipulation. See [here](https://en.cppreference.com/w/cpp/filesystem/path). 

@shaomeng and I had a discussion about moving to C++17 a few weeks ago. Using C++17 features may not be something we want to look at right now. On the other hand, doing the experiment to see if std::filesystem is supported on all of our platforms should not be too hard. At the very least it would be worth looking at what is in std::filesystem so you know what is coming.

cc @StasJ

Reply:
Ok, I'll look into it and report back.  Hopefully we can bump to c++17.

Reply:
I'm already using c++17 in my compression code and there are some very nice things about it.

Reply:
That's encouraging. Hopefully they work on all of our supported platforms. We shouldn't make the jump until we know this for sure. To be clear, I'm not asking @sgpearse to use C++17 if the timing isn't right. But at very least we should using are existing platform-portable path handing tools, with the long term goal of being able to replace these with something supported by the STL.

Reply:
I would be concerned about Windows support with C++17 features

Reply:
Currently using the WASP library's FileUtils for this.

Reply:
@sgpearse please use FileUtils::Dirname to get the directory path here:

https://github.com/NCAR/VAPOR/blob/527fa55a484b2af30478774dbbe6ecea8e07c00d/lib/vdc/BOVCollection.cpp#L97

and use FileUtils::JoinPaths to concatenate paths here:

https://github.com/NCAR/VAPOR/blob/527fa55a484b2af30478774dbbe6ecea8e07c00d/lib/vdc/BOVCollection.cpp#L261

cc @StasJ